### DISCUSSION TITLE
App Store Connect - Provided value is not a valid PEM encoded private key

### DISCUSSION THREAD

**Discussion by vmlb13** [codemagic-ci-cd/codemagic-docs#2637] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I'm getting this error in my pre build script, I tried to change the private key several times and I also printed the private key from the environment variable, the content it's there and in the right format, but the issue seems to stay.

Error:
**app-store-connect get-latest-app-store-build-number: error: argument --private-key: Provided value is not a valid PEM encoded private key
Failed to get the latest build number from App Store Connect.**

Script:

```
#!/bin/sh

# Check if PACKAGE_NAME is set
if [ -z "$PACKAGE_NAME" ]; then
  echo "PACKAGE_NAME is not set. Please set the PACKAGE_NAME environment variable."
  exit 1
fi

# Check if APP_APPLE_ID is set
if [ -z "$APP_APPLE_ID" ]; then
  echo "APP_APPLE_ID is not set. Please set the APP_APPLE_ID environment variable."
  exit 1
fi

# Check if APP_STORE_CONNECT_ISSUER_ID is set
if [ -z "$APP_STORE_CONNECT_ISSUER_ID" ]; then
  echo "APP_STORE_CONNECT_ISSUER_ID is not set. Please set the APP_STORE_CONNECT_ISSUER_ID environment variable."
  exit 1
fi

# Check if APP_STORE_CONNECT_KEY_IDENTIFIER is set
if [ -z "$APP_STORE_CONNECT_KEY_IDENTIFIER" ]; then
  echo "APP_STORE_CONNECT_KEY_IDENTIFIER is not set. Please set the APP_STORE_CONNECT_KEY_IDENTIFIER environment variable."
  exit 1
fi

# Check if APP_STORE_CONNECT_PRIVATE_KEY is set
if [ -z "$APP_STORE_CONNECT_PRIVATE_KEY" ]; then
  echo "APP_STORE_CONNECT_PRIVATE_KEY is not set. Please set the APP_STORE_CONNECT_PRIVATE_KEY environment variable."
  exit 1
fi

LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number \
                      "$APP_APPLE_ID" \
                      --issuer-id "$APP_STORE_CONNECT_ISSUER_ID" \
                      --key-id "$APP_STORE_CONNECT_KEY_IDENTIFIER" \
                      --private-key "$APP_STORE_CONNECT_PRIVATE_KEY" \
                      --disable-jwt-cache)

if [ $? -ne 0 ] || [ -z "$LATEST_BUILD_NUMBER" ]; then
  echo "Failed to get the latest build number from App Store Connect."
  echo "Setting build number to 1 for first-time submission."
  LATEST_BUILD_NUMBER=0
fi
```

### Expected results

It's supposed to get the latest build number from the app store

### Actual results

Error:
**app-store-connect get-latest-app-store-build-number: error: argument --private-key: Provided value is not a valid PEM encoded private key
Failed to get the latest build number from App Store Connect.**

### Build id (optional)

6699350e3c6f7f72927a91f0

**Comment by dtrdic** [comment]
hello @vmlb13 I see you have overcome the previous error by removing the script entirely and adding build arguments with the auto-incremented built-in environment variable `--build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=$PROJECT_BUILD_NUMBER`, which are then used for versioning your app.

Now I'm noticing a different error on your latest build

> The apk has permissions that require a privacy policy set for the app, e.g: android.permission.CAMERA

This could also happen if you haven't provided any privacy policy to your app in the Google Play Developer Console. 
Please check [this discussion](https://www.googlecloudcommunity.com/gc/AppSheet-Q-A/The-APK-file-or-Android-App-Bundle-uses-permissions-that-require/td-p/357502) for reference

**Comment by vmlb13** [comment]
Hi @dtrdic, yes I changed the approach for the automatic build versions from code magic.
I'm aware of the issue because of privacy policy, thanks for just clarifying.
